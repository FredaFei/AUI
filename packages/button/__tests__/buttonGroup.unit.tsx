import * as renderer from 'react-test-renderer'
import React from 'react'
import {shallow, mount} from 'enzyme'

import Button from '../button'
import ButtonGroup from '../buttonGroup'

describe('测试 ButtonGroup 以下行为', () => {
  it('ButtonGroup 存在', () => {
    const json = renderer.create(
      <ButtonGroup>
        <Button icon="left"></Button>
        <Button>1</Button>
        <Button icon="right"></Button>
      </ButtonGroup>
    ).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('ButtonGroup 子组件不是Button组件时', () => {
    const wrapper = mount(
      <ButtonGroup>hello</ButtonGroup>
    )
    expect(wrapper.find('.am-buttonGroup').text()).toEqual('')
  })
  it('ButtonGroup 设置 自定义类名', () => {
    const wrapper = shallow(
      <ButtonGroup className="hello">
        <Button icon="left"></Button>
        <Button>1</Button>
        <Button icon="right"></Button>
      </ButtonGroup>
    )
    expect(wrapper.find('.am-buttonGroup').hasClass('hello')).toBe(true)
  })
  it('ButtonGroup 设置 自定义样式', () => {
    const wrapper = renderer.create(
      <ButtonGroup className="hello" style={{borderColor: 'red'}}>
        <Button icon="left"></Button>
        <Button>1</Button>
        <Button icon="right"></Button>
      </ButtonGroup>
    ).toJSON()
    expect(wrapper).toMatchSnapshot()
  });
})