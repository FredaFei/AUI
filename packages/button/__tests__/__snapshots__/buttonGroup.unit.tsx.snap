// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 ButtonGroup 以下行为 ButtonGroup 存在 1`] = `
<div
  className="am-buttonGroup "
  data-role="ButtonGroup"
>
  <button
    className="am-button-wrapper "
    data-role="Button"
    onClick={[Function]}
  >
    <div
      className="am-button-body icon-left"
    >
      <svg
        className="am-icon icon-name"
        style={Object {}}
      >
        <use
          xlinkHref="#left"
        />
      </svg>
      <span
        className="am-button-content"
      />
    </div>
  </button>
  <button
    className="am-button-wrapper "
    data-role="Button"
    onClick={[Function]}
  >
    <span
      className="am-button-content"
    >
      1
    </span>
  </button>
  <button
    className="am-button-wrapper "
    data-role="Button"
    onClick={[Function]}
  >
    <div
      className="am-button-body icon-left"
    >
      <svg
        className="am-icon icon-name"
        style={Object {}}
      >
        <use
          xlinkHref="#right"
        />
      </svg>
      <span
        className="am-button-content"
      />
    </div>
  </button>
</div>
`;

exports[`测试 ButtonGroup 以下行为 ButtonGroup 设置 自定义样式 1`] = `
<div
  className="am-buttonGroup hello"
  data-role="ButtonGroup"
  style={
    Object {
      "borderColor": "red",
    }
  }
>
  <button
    className="am-button-wrapper "
    data-role="Button"
    onClick={[Function]}
  >
    <div
      className="am-button-body icon-left"
    >
      <svg
        className="am-icon icon-name"
        style={Object {}}
      >
        <use
          xlinkHref="#left"
        />
      </svg>
      <span
        className="am-button-content"
      />
    </div>
  </button>
  <button
    className="am-button-wrapper "
    data-role="Button"
    onClick={[Function]}
  >
    <span
      className="am-button-content"
    >
      1
    </span>
  </button>
  <button
    className="am-button-wrapper "
    data-role="Button"
    onClick={[Function]}
  >
    <div
      className="am-button-body icon-left"
    >
      <svg
        className="am-icon icon-name"
        style={Object {}}
      >
        <use
          xlinkHref="#right"
        />
      </svg>
      <span
        className="am-button-content"
      />
    </div>
  </button>
</div>
`;
